test/junit/scala/collection/immutable/StreamTest.scala

```diff
@@ -15,4 +18,58 @@ class StreamTest {
     assertTrue(Stream(1,2,3,4,5).filter(_ < 4) == Seq(1,2,3))
     assertTrue(Stream(1,2,3,4,5).filterNot(_ > 4) == Seq(1,2,3,4))
   }
+
+  /** Test helper to verify that the given Stream operation allows
+    * GC of the head during processing of the tail.
+    */
+  def assertStreamOpAllowsGC[B](op: (=> Stream[Int], Int => B) => Any, f: Int => B): Unit = {
+    val msgSuccessGC = "GC success"
+    val msgFailureGC = "GC failure"
+
+    val ref = WeakReference( Stream.from(1).take(500) ) // wait about 5 seconds to fail
+
+    def gcAndThrowIfCollected(n: Int): B = {
+      System.gc()
+      Thread.sleep(10)
+      if (ref.get.isEmpty) throw new RuntimeException(msgSuccessGC)
+      assertTrue(n >= 1) // protect from JIT optimizations
```

Comment:
Why not try removing it and see if it still works.  It's really helpful to have all the code either have an obvious purpose, or a comment explaining why this particular thing is necessary.  "Someone suggested this" is a little dodgy either way.